modules/mri_violations/php/resolved_violations.class.inc

```diff
@@ -205,17 +205,19 @@ class Resolved_Violations extends \NDB_Menu_Filter_Form
                               time_run
                    )
                 ) as hash,
-                mri_protocol_violated_scans.ID as join_id,                
+                mpvs.ID as join_id,                
                 p.CenterID as Site,
                 violations_resolved.Resolved as Resolved
-            FROM mri_protocol_violated_scans 
+            FROM mri_protocol_violated_scans AS mpvs
             LEFT JOIN violations_resolved
-            ON (violations_resolved.ExtID=mri_protocol_violated_scans.ID 
+            ON (violations_resolved.ExtID=mpvs.ID 
             AND violations_resolved.TypeTable='mri_protocol_violated_scans')
             LEFT JOIN candidate c
-            ON (mri_protocol_violated_scans.CandID = c.CandID)
+            ON (mpvs.CandID = c.CandID)
             LEFT JOIN session s
-            ON (mri_protocol_violated_scans.CandID = s.CandID)
+            ON (SUBSTRING_INDEX(mpvs.PatientName,'_',-1) = s.Visit_label 
```

Comment:
I am not sure it's safe to assume that the patient name will always contain the visit label. If we change this convention some day this SQL statement might cease to work...Why not join with `tarchive` and then with `session` since both tables are linked via a session ID?

Reply:
Is this still relevant, given your "approved" review?

Reply:
@driusan Not anymore